byzcoin/proto.go

```diff
@@ -198,6 +200,9 @@ type Instruction struct {
 	// Signatures that are verified using the Darc controlling access to
 	// the instance.
 	Signatures [][]byte
+	// version is a private field that can allow an instruction to be passed
```

Comment:
I'm not a fan of passing the version as a private field in this struct. Is it to prevent adding new methods that take the version-number as argument?

If you keep it, I would add an additional comment that states something like:
```
This field must be the last field of the struct, so that the protobuf-library enumerates the fields correctly.
```

Reply:
Yes, that means a lot of function prototype changes and some part of the client code that would need to take this into account so for me it's better to do it like that. It can still be improved for v4 with better abstractions.

Noted for the comment.